### PR TITLE
Added publishable draft on patterns discussed so far

### PR DESCRIPTION
**Pull request by BaardBouvet**

### CODE REVIEW

File: patterns.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+Fetch more data based on some input source, requires rescan all the time, perhaps partial rescan could work?
+
+Recreate best effort history from a source
+------------------------------------------
+We become master, pipe should be durable.
+
+Make periodic entities from a versioned history
+-----------------------------------------------
+We do that in Sesam Sesam from the datasets we read Sesam Portal that only has a current view of the world in
+latest.
```
Comment (JonasAls89) [inline patterns.rst:39]: this sentence must be reformulated. Makes little sense as is.

File: patterns.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+Sporadic empty response
+-----------------------
+Source sometimes produces an empty array for some reason (during restarts, authenticaion problems, etc). Use circuit breakers.
```
Comment (JonasAls89) [inline patterns.rst:43]: add ref:`circuit breakers <concepts-circuit-breakers>` link here maybe?

File: patterns.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+Enrich patterns
+===============
+
+Extract foreign references as separate datatypes
+------------------------------------------------
+Don't make NIs to stuff that is outside your control, keep the namespace local to the system. Extract the
+properties to new separate datatypes. If you don't have them as objects you can't merge them with the same concept from
+other sources. Time is not a good candidate for NI. Postnummer is an example. If you make a ni make it reference your
```
Comment (JonasAls89) [inline patterns.rst:52]: Postnumber

File: patterns.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+Useful to pick out relevant subsets from the globals later. Add data type property (rdf:type or $type).
+
+Flattening complex objects
+--------------------------
+Gives you flat structures. Can be easier to work with.
+
+Splitting out lists of sub objects
+----------------------------------
+Aggregate objects, are the sub objects part of the parent or can they live on their own? Use create-child and
+emit_children.
```
Comment (JonasAls89) [inline patterns.rst:66]: ref links her og.

**Review by JonasAls89 — CHANGES_REQUESTED** [review]
Some suggestions.

File: patterns.rst [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+Connect patterns
+================
+
+Washing data
+------------
+Should be added as new properties, you might need the dirty data.
+
+Manual merge
+------------
+Hardhoded dataset with manually connected ids, could also be an external source with manual input. Linking table.
```
Comment (JonasAls89) [inline patterns.rst:78]: Type? Hardcoded dataset..

**Review by JonasAls89 — CHANGES_REQUESTED** [review]

File: patterns.rst (hunk 5 above)
Comment (JonasAls89) [reply patterns.rst:78]: typo*

Comment (BaardBouvet) [reply patterns.rst:78]: You've fixed this in a later PR, so I'll ignore this for now.

**Review by branislavjenco — APPROVED** [review]

**Pull request merged**